src/formatter.rs

```diff
@@ -1,108 +1,230 @@
 //! Functions related to formatting and printing lines from a `Tokenizer`.
 
-use std::io::{BufRead, Write};
-
-use ansi_term::{ANSIString, ANSIStrings};
-use log::debug;
-
-use crate::config::Config;
-use crate::error::TealdeerError::{self, WriteError};
-use crate::tokenizer::Tokenizer;
+use crate::extensions::FindFrom;
 use crate::types::LineType;
 
-fn highlight_command<'a>(
-    command: &'a str,
-    example_code: &'a str,
-    config: &Config,
-    parts: &mut Vec<ANSIString<'a>>,
-) {
-    let mut code_part_end_pos = 0;
-    while let Some(command_start) = example_code[code_part_end_pos..].find(&command) {
-        let code_part = &example_code[code_part_end_pos..code_part_end_pos + command_start];
-        parts.push(config.style.example_code.paint(code_part));
-        if code_part_end_pos == 0 {
-            // Only highlight command names at the start of the line ...
-            parts.push(config.style.command_name.paint(command));
-        } else {
-            let char_before_command = example_code
-                .chars()
-                .nth(code_part_end_pos + command_start - 1);
-            if char_before_command.filter(|c| c.is_whitespace()).is_some() {
-                // ... or when preceded by a whitespace character.
-                parts.push(config.style.command_name.paint(command));
-            } else {
-                parts.push(config.style.example_code.paint(command));
-            }
-        }
+use log::debug;
 
-        code_part_end_pos += command_start + command.len();
-    }
-    parts.push(
-        config
-            .style
-            .example_code
-            .paint(&example_code[code_part_end_pos..]),
-    );
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+/// Represents a snippet from a page from a specific highlighting class.
+pub enum HighlightingSnippet<'a> {
+    CommandName(&'a str),
+    Variable(&'a str),
+    NormalCode(&'a str),
+    Description(&'a str),
+    Text(&'a str),
+    Linebreak,
 }
```

Comment:
I think the name of this enum is not ideal, since the type will be used even when disabling any syntax highlighting.

Maybe call it `OutputSnippet` instead? (Unless you have a better idea. Not a big fan of that one either, but I couldn't come up with something better.)

Reply:
I feel like `OutputSnippet` implies too much what it is supposed to be used for, not what it actually contains. What about something like `PageSnippet`?

Reply:
Yep, that sounds good!